### PR TITLE
Clean up Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs

### PR DESCRIPTION
**Pull request by Bond-009**
Fixed all warnings in Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs

### CODE REVIEW

File: Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs [hunk 1]
```diff
@@ -1644,22 +1611,20 @@ private void DeleteLibraryItemForTimer(TimerInfo timer)
...
+                    new DeleteOptions
+                    {
+                        DeleteFileLocation = true
+                    },
+                    true);
             }
             else
             {
-                try
+                if (File.Exists(timer.RecordingPath))
```
Comment (JustAMan) [inline Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs]: I actually don't like when you remove exception handling, what if file exists but we cannot delete it for whatever reason (no permissions, read-only file system, i/o failure)?

File: Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs [hunk 2]
```diff
@@ -1776,44 +1730,16 @@ private async Task SaveRecordingImage(string recordingPath, LiveTvProgram progra
...
-            }
+                ImageType.Primary => program.IsSeries ? Path.GetFileNameWithoutExtension(recordingPath) + "-thumb" : "poster",
+                ImageType.Logo => "logo",
+                ImageType.Thumb => program.IsSeries ? Path.GetFileNameWithoutExtension(recordingPath) + "-thumb" : "landscape",
+                ImageType.Backdrop => "fanart",
+                _ => null
+            };
 
-            if (string.IsNullOrWhiteSpace(imageSaveFilenameWithoutExtension))
+            if (imageSaveFilenameWithoutExtension == null)
```
Comment (JustAMan) [inline Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs]: why?

File: Emby.Server.Implementations/LiveTv/EmbyTV/EpgChannelData.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        public ChannelInfo GetChannelByName(string name)
+        {
+            ChannelsByName.TryGetValue(name, out var result);
+
+            return result;
+        }
+
+        public static string NormalizeName(string value)
+        {
+            return value.Replace(" ", string.Empty, StringComparison.Ordinal).Replace("-", string.Empty, StringComparison.Ordinal);
```
Comment (JustAMan) [inline Emby.Server.Implementations/LiveTv/EmbyTV/EpgChannelData.cs]: I wonder why we do this... I can imagine having two separate channels named like "TV-1" and "TV1" being two different channels

File: MediaBrowser.MediaEncoding/Subtitles/SubtitleEncoder.cs [hunk 4]
```diff
@@ -126,7 +126,7 @@ async Task<Stream> ISubtitleEncoder.GetSubtitles(BaseItem item, string mediaSour
                 throw new ArgumentNullException(nameof(mediaSourceId));
             }
 
-            var mediaSources = await _mediaSourceManager.GetPlayackMediaSources(item, null, true, false, cancellationToken).ConfigureAwait(false);
+            var mediaSources = await _mediaSourceManager.GetPlaybackMediaSources(item, null, true, false, cancellationToken).ConfigureAwait(false);
```
Comment (JustAMan) [inline MediaBrowser.MediaEncoding/Subtitles/SubtitleEncoder.cs]: this typo always jarred me, but I was too lazy to fix it lol

File: Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs]: The caller already catches it
```cs
                try
                {
                    DeleteLibraryItemForTimer(timer);
                }
                catch (Exception ex)
                {
                    _logger.LogError(ex, "Error deleting recording");
                }
```

File: Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs (hunk 2 above)
Comment (Bond-009) [reply Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs]: It can't possibly be empty or whitespace only

**Comment by JustAMan** [comment]
@Bond-009 am I right that CI now fails because Azure updated stuff on their end?.. or is it us calling `dotnet publish` incorrectly?

**Comment by Bond-009** [comment]
@JustAMan Nope, this is my bad, I'll fix it

**Comment by EraYaN** [comment]
This might need a rebase on current master to fix the addition of the CA1056 rule to the rule set (master already has it as "None") and this PR does not alter that one.

**Pull request merged**